### PR TITLE
Better support for Deconv and Crop layers

### PR DESCRIPTION
**Pull request by akshaychawla**
1. support for converting deconvolution and crop layers from caffe to keras. Very useful if one is working with image segmentation models. 
2. sample deploy.prototxt and tests are added in keras/caffe/models/segmentation/ to check the implementation. 
3. A sample output is shown in keras/caffe/models/segmentation/mini_proof.png.
4. Slight change in caffe_to_keras function in convert.py. In case input_dim from config.input_dim[:] is an empty list, the dimensions are sourced from the "data" layer, have a look at the code once. 
5. Softmax now supports inputs of type (Batchsize, classes, height, width) which occurs in image segmentation networks and converts it to the format (Batchsize, height*width, classes) which can be given to standard softmax activation in Keras. 
6. When copying over weights, Deconvolution weights are copied in the same way as convolution weights. I added a zeros vector to weights_b in case has_bias is False.

### CODE REVIEW

**Comment by MarcBS** [comment]
I tested it and it is working great!

**Comment by akshaychawla** [comment]
Yay :) I've also been working towards adding a Scale layer which is used in voc-fcn8s-allatonce (https://github.com/shelhamer/fcn.berkeleyvision.org/tree/master/voc-fcn8s-atonce), will make a PR when that's ready
